content/blog/how-to-use-dscc-api-and-ansible-to-collect-the-storage-configuration.md

```diff
@@ -0,0 +1,223 @@
+    headers:
+      Authorization: "{{ token }}"
+      Content-Type: "application/json"
+    method: "{{ method }}"
+    validate_certs: false
+    status_code: [200, 201, 202, 401, 404]
+  register: result
+  when: body is not defined
+
+- name: Set result status
+  ansible.builtin.set_fact:
+    status: "{{ result.status }}"
+    tmpres: "{{ result }}"
+  when: body is not defined
+
+- name: DSCC Command with body {{request_uri}}
+  ansible.builtin.uri:
+    url: "{{ base_url }}{{ request_uri }}"
+    headers:
+      Authorization: "{{ token }}"
+      Content-Type: "application/json"
+    method: "{{ method }}"
+    body_format: json
+    body: "{{ body | to_json }}"
+    validate_certs: false
+    status_code: [200, 201, 202, 400, 401, 404]
+  register: result2
+  when: body is defined
+
+- name: Set result status
+  ansible.builtin.set_fact:
+    status: "{{ result2.status }}"
+    tmpres: "{{ result2 }}"
+  when: body is defined
+
+- name: Set response when status in [200, 201, 202, 401]
+  ansible.builtin.set_fact:
+    response: "{{ tmpres }}"
+  when: status in ['200', '201', '202','401']
+
+- name: Undefine Response when status not in [200...]
+  ansible.builtin.set_fact:
+    response: ""
+  when: status not in ['200', '201', '202','401']
+```
+
+You can see, that it first retrieves the stored access token, then checks that the method and the request URI is available. Next it issues the API call either with or without a call body before the possible response status is checked and the call response is set. 
+
+# System Configuration capture 
+
+The complete workflow of the DSCC data capturing is shown in the following flow diagram. First the list of connected storage arrays is compiled and stored in a dictionary. Next the playbook will loop through the storage array dictionary in order to capture the details of each connected storage array (this includes looping through all associated links of a storage array and the gathering of all volumes that are defined on the storage array). Afterwards the host group and host details are captured and stored too. 
+
+![](/img/capturestorage-flowdiagram.png "Capture Storage System Flow Diagram")
+
+This system configuration capture flow chart can now be implemented using the above mentioned basic task in combination with the correct request URIs and request bodies. You can see in the example below, that the playbook first gets the list of storage arrays (request uri: /api/v1/storage-systems) and if the command returns a status code of 401 (i.e. unauthorized access) it repeats the same call after retrieving a refreshed access token (that is the difference between the DSCC-API-Call.yaml and the DSCC-API-401.yaml playbook).  After successfully retrieving the system list, a system dictionary is populated first, followed by looping through the dictionary (Loop-Systems.yml playbook) and storing the system configuration information. Afterwards, the host group and hosts details are retrieved and stored. 
```

Comment:
I'd recommend breaking up the second sentence into a couple of simpler sentences. For example:

You can see in the example below that the playbook first gets the list of storage arrays (request uri:/api/v1/storage-systems). If the command returns a status code of 401 (i.e. unauthorized access), it repeats the same call after retrieving a refreshed access token (this is the difference between the DSCC-API-Call.yaml and the DSCC-API(-401.yaml playbook).